smartsim/settings/mpirunSettings.py

```diff
@@ -24,318 +24,14 @@
-        """Settings to run job with ``mpiexec`` command (OpenMPI)
-
-        Note that environment variables can be passed with a None
-        value to signify that they should be exported from the current
-        environment
-
-        Any arguments passed in the ``run_args`` dict will be converted
-        into ``mpiexec`` arguments and prefixed with ``--``. Values of
-        None can be provided for arguments that do not have values.
-
-        :param exe: executable
-        :type exe: str
-        :param exe_args: executable arguments, defaults to None
-        :type exe_args: str | list[str], optional
-        :param run_args: arguments for run command, defaults to None
-        :type run_args: dict[str, str], optional
-        :param env_vars: environment vars to launch job with, defaults to None
-        :type env_vars: dict[str, str], optional
-        """
-        super().__init__(exe, exe_args, "mpiexec", run_args, env_vars, **kwargs)
-
-        completed_process = subprocess.run(
-            [self.run_command, "-V"], capture_output=True
-        )  # type: subprocess.CompletedProcess
-        version_statement = completed_process.stdout.decode()
-
-        if not re.match(r"mpiexec\s\(OpenRTE\)\s4.\d+.\d+", version_statement):
-            logger.warning("Non-OpenMPI implementation of `mpiexec` detected")
-
-
-class OrterunSettings(_OpenMPISettings):
-    def __init__(self, exe, exe_args=None, run_args=None, env_vars=None, **kwargs):
-        """Settings to run job with ``orterun`` command (OpenMPI)
-
-        Note that environment variables can be passed with a None
-        value to signify that they should be exported from the current
-        environment
-
-        Any arguments passed in the ``run_args`` dict will be converted
-        into ``orterun`` arguments and prefixed with ``--``. Values of
-        None can be provided for arguments that do not have values.
-
-        :param exe: executable
-        :type exe: str
-        :param exe_args: executable arguments, defaults to None
-        :type exe_args: str | list[str], optional
-        :param run_args: arguments for run command, defaults to None
-        :type run_args: dict[str, str], optional
-        :param env_vars: environment vars to launch job with, defaults to None
-        :type env_vars: dict[str, str], optional
-        """
-        super().__init__(exe, exe_args, "orterun", run_args, env_vars, **kwargs)
-
-        completed_process = subprocess.run(
-            [self.run_command, "-V"], capture_output=True
-        )  # type: subprocess.CompletedProcess
-        version_statement = completed_process.stdout.decode()
-
-        if not re.match(r"orterun\s\(OpenRTE\)\s4.\d+.\d+", version_statement):
-            logger.warning("Non-OpenMPI implementation of `orterun` detected")
+simplefilter("once", DeprecationWarning)
+warn(
+    "mpirunSettings will be deprecated; use mpiSettings instead.",
+    DeprecationWarning
```

Comment:
add `stacklevel=2`?

Reply:
At this point, you're allowed to yell at me, since I reneged on my promise to do this earlier